pyvista/plotting/_plotting.py

```diff
@@ -0,0 +1,147 @@
+        :ref:`shading_example`.
+    feature_angle : float, optional
+        Angle to consider an edge a sharp edge.
+
+    Returns
+    -------
+    pyvista.PolyData
+        Always a surface as we need to compute point normals.
+
+    """
+    # extract surface if mesh is exterior
+    if not isinstance(mesh, pyvista.PolyData):
+        grid = mesh
+        mesh = grid.extract_surface()
+        # remap scalars
+        if scalars is not None:
+            ind = mesh.point_data['vtkOriginalPointIds']
+            scalars = np.asarray(scalars[ind])
+    if texture:
+        _tcoords = mesh.active_t_coords
+
+    if split_sharp_edges:
+        # we must track the original IDs
+        indices = np.arange(mesh.n_points, dtype=np.int32)
+        mesh.point_data['__orig_ids__'] = indices
+        mesh = mesh.compute_normals(
+            cell_normals=False,
+            split_vertices=True,
+            feature_angle=feature_angle,
+        )
+        if scalars is not None:
+            ind = mesh.point_data['__orig_ids__']
+            scalars = np.asarray(scalars)[ind]
+    else:
+        mesh.compute_normals(cell_normals=False, inplace=True)
+
+    if texture:
+        mesh.active_t_coords = _tcoords
+
+    return mesh, scalars
+
+
+def process_opacity(mesh, opacity, preference, n_colors, scalars, use_transparency):
+    """Process opacity.
+
+    This function accepts an opacity string or array and always
+    returns an array that can be applied to a dataset for plotting.
+
+    Parameters
+    ----------
+    mesh : pyvista.DataSet
+        Dataset to process the opacity for.
+    opacity : str, numpy.ndarray
+        String or array.  If string, must be a cell or point data array.
+        preference : str, optional
+            When ``mesh.n_points == mesh.n_cells``, this parameter
+            sets how the scalars will be mapped to the mesh.  Default
+            ``'points'``, causes the scalars will be associated with
+            the mesh points.  Can be either ``'points'`` or
+            ``'cells'``.
+    n_colors : int, optional
+        Number of colors to use when displaying the opacity.
+    scalars : numpy.ndarray, optional
+        Dataset scalars.
+    use_transparency : bool, optional
+        Invert the opacity mappings and make the values correspond
+        to transparency.
+
+    Returns
+    -------
+    _custom_opac : bool
+        If using custom opacity.
+    opacity : numpy.ndarray
+        Array containing the opacity.
+
+    """
+    _custom_opac = False
+    if isinstance(opacity, str):
+        try:
+            # Get array from mesh
+            opacity = get_array(mesh, opacity,
+                                preference=preference, err=True)
+            if np.any(opacity > 1):
+                warnings.warn("Opacity scalars contain values over 1")
+            if np.any(opacity < 0):
+                warnings.warn("Opacity scalars contain values less than 0")
+            _custom_opac = True
+        except:
```

Comment:
We should use a more targeted except here.

Reply:
Fixed in e11863950a240be285b5e3cb691736e83c12d5ed